docs/spec/auth/subkeys_proposal.md

```diff
@@ -0,0 +1,156 @@
+# Subkeys ADR
+
+## Abstract
+
+Currently, a `StdAccount` has only one public key assoiciated with it, and all transactions that are made with that key must be signed by that one public key. This is problematic because the security level required by different types of transactions executable by an account may be different. For example, an account may want to keep the key that has the ability to do `SendMsg`s in cold storage, but wants to keep a key that can vote on governance proposals on their phone.
+
+For this reason, we introduce the concept of SubKeys.  Accounts should be able to have multiple public keys, each of which can have different permissions.  The main public key of account can provision new subkeys with specific msg routes who they are allowed to sign for.  If a subkey tries to sign a tx with a msg type it is not permitted to use, the tx will be rejected.
+
+One complexity in this system is that all txs, regardless of their msg type, need to be able to tx fees. This means that even your lowest security subkey, if compromised, could drain your account by using your entire balance as the tx fee for a transaction. To resolve this, we add a notion of FeeAllowances.  The master pubkey along with provisioning subkeys with msg types they are allowed to use, also provisions them with a daily allowance of tokens that they are allowed to spend to pay transaction fees.  For example, if SubKey A has a daily fee allowance of 1atom, it can spend up to 1atom on transaction fees in a 24 hour window.  If it does not use its atom in a 24 hour window, their "allowance" does not accumulate, it still stays at 1atom in any 24 hour window.
```

Comment:
```suggestion
One complexity in this system is that all txs, regardless of their msg type, need to be able to send tx fees. This means that even your lowest security subkey, if compromised, could drain your account by using your entire balance as the tx fee for a transaction. To resolve this, we add a notion of FeeAllowances.  The master pubkey, along with provisioning subkeys with msg types they are allowed to use, also provisions them with a daily allowance of tokens that they are allowed to spend to pay transaction fees.  For example, if SubKey A has a daily fee allowance of 1atom, it can spend up to 1atom on transaction fees in a 24 hour window.  If it does not use its atom in a 24 hour window, the "allowance" does not accumulate, it still stays at 1atom in any 24 hour window.
```